fitbenchmarking/fitting/controllers/gsl_controller.py

```diff
@@ -0,0 +1,188 @@
+            self._solver = multifit_nlin.lmsder(mysys, self._n, self._p)
+        elif self.minimizer=='lmder':
+            self._solver = multifit_nlin.lmder(mysys, self._n, self._p)
+        elif self.minimizer=='simplex':
+            self._solver = multiminimize.nmsimplex(mysys, self._p)
+        elif self.minimizer=='simplex2':
+            self._solver = multiminimize.nmsimplex2(mysys, self._p)
+        elif self.minimizer=='conjugate_pr':
+            self._solver = multiminimize.conjugate_pr(mysys, self._p)
+        elif self.minimizer=='conjugate_fr':
+            self._solver = multiminimize.conjugate_fr(mysys, self._p)
+        elif self.minimizer=='bfgs':
+            self._solver = multiminimize.bfgs(mysys, self._p)
+        elif self.minimizer=='bfgs2':
+            self._solver = multiminimize.bfgs2(mysys, self._p)
+        elif self.minimizer=='steepest_descent':
+            self._solver = multiminimize.steepest_descent(mysys, self._p)
+
+        # Set up initialization parameters
+        # 
+        # These have been chosen to be consistent with the parameters
+        # used in Mantid.
+        self._initial_steps = 1.0 * numx.array(np.ones(self._p))
+        self._step_size = 0.1
+        self._tol = 1e-4
+        self._gradient_tol = 1e-3        
+        self._abserror = 1e-4
+        self._relerror = 1e-4
+        self._maxits    = 500
+
+        if self.minimizer in self._residual_methods:
+            self._solver.set(self._pinit)
+        elif self.minimizer in self._function_methods_no_jac:
+            self._solver.set(self._pinit, self._initial_steps)
+        else:
+            self._solver.set(self._pinit, self._step_size, self._tol)
+
+    def fit(self):
+        """ 
+        Run problem with GSL
+        """
+        self.success = False
+
+        for iter in range(self._maxits):
+            status = self._solver.iterate()
+            # check if the method has converged
+            if self.minimizer in self._residual_methods:
+                x  = self._solver.getx()
+                dx = self._solver.getdx()
+                status = multifit_nlin.test_delta(dx, x,
+                                                  self._abserror,
+                                                  self._relerror)
+            elif self.minimizer in self._function_methods_no_jac:
+                simplex_size = self._solver.size()
+                status = multiminimize.test_size(simplex_size,
+                                                 self._abserror)
+            else: # must be in function_methods_with_jac
+                gradient = self._solver.gradient()
+                status = multiminimize.test_gradient(gradient,
+                                                     self._gradient_tol)
+            if status == errno.GSL_SUCCESS:
+                self.success = True
+                break
+            elif status != errno.GSL_CONTINUE:
+                raise ValueError("GSL couldn't find a solution")
```

Comment:
I think this should be a RuntimeError rather than ValueError.
```suggestion
                raise RuntimeError("GSL couldn't find a solution")
```